keystorage/aws/src/main/java/tech/pegasys/signers/aws/CacheableAwsSecretsManager.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.aws;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+
+public class CacheableAwsSecretsManager {
+
+  private static final Cache<String, AwsSecretsManager> awsSecretsManagerCache =
```

Comment:
Still don't like the static field here. Think it's safe because of the concurrent hashmap but we will still need be careful how use the CacheableAwsSecretsManager. E.g. what happens if a reload is happening at the same time as the initial key loading do we need to be careful when we call clearCache?